Marlin/planner.cpp

```diff
@@ -1013,26 +1015,42 @@ void Planner::_buffer_line(const float &a, const float &b, const float &c, const
   }
 
   // Compute and limit the acceleration rate for the trapezoid generator.
-  float steps_per_mm = block->step_event_count / block->millimeters;
+  float steps_per_mm = block->step_event_count * inverse_millimeters;
   uint32_t accel;
   if (!block->steps[X_AXIS] && !block->steps[Y_AXIS] && !block->steps[Z_AXIS]) {
     // convert to: acceleration steps/sec^2
     accel = ceil(retract_acceleration * steps_per_mm);
   }
   else {
-    #define LIMIT_ACCEL(AXIS) do{ \
-      if (max_acceleration_steps_per_s2[AXIS] < (accel * block->steps[AXIS]) / block->step_event_count) \
-        accel = (max_acceleration_steps_per_s2[AXIS] * block->step_event_count) / block->steps[AXIS]; \
+    #define LIMIT_ACCEL_32(AXIS) do{ \
+      if (block->steps[AXIS]) { \
+        const uint32_t comp = max_acceleration_steps_per_s2[AXIS] * block->step_event_count; \
+        if (accel * block->steps[AXIS] > comp) accel = comp / block->steps[AXIS]; \
+      } \
+    }while(0)
+
+    #define LIMIT_ACCEL_64(AXIS) do{ \
+      if (block->steps[AXIS]) { \
+        const uint64_t comp = (uint64_t)max_acceleration_steps_per_s2[AXIS] * (uint64_t)block->step_event_count; \
+        if ((uint64_t)accel * (uint64_t)block->steps[AXIS] > comp) accel = comp / (uint64_t)block->steps[AXIS]; \
+      } \
```

Comment:
Thanks for the numbers, now I can skip the test myself :)
Hard to believe that a 64bit division should be faster than 32bit. Is it possible that the compiler does some optimization due to the fixed values of the two variables?

The reason why I used uint64_t is because Sailfish is using it. I never used them before, so I thought it might be faster than float if they use it istead of float. Looks like that wasn't the best idea.
I used an old mathematical program to write down the acceleration calculation so I can play with the values and see where problems might occur. I will use that to see if there is a safe way to use a bitshift today..